jmetal-core/src/main/java/org/uma/jmetal/operator/impl/crossover/PMXCrossover.java

```diff
@@ -145,10 +146,23 @@ public void setCrossoverProbability(double crossoverProbability) {
         offspring.get(0).setVariableValue(i, n1);
         offspring.get(1).setVariableValue(i, n2);
       }
-    }
+}
 
-    return offspring;
-  }
+private void interchange(List<PermutationSolution<Integer>> parents, List<PermutationSolution<Integer>> offspring,
+		int cuttingPoint1, int cuttingPoint2, int[] replacement1, int[] replacement2) {
+	for (int i = cuttingPoint1; i <= cuttingPoint2; i++) {
+        offspring.get(0).setVariableValue(i, parents.get(1).getVariableValue(i));
+        offspring.get(1).setVariableValue(i, parents.get(0).getVariableValue(i));
+
+        replacement1[parents.get(1).getVariableValue(i)] = parents.get(0).getVariableValue(i) ;
+        replacement2[parents.get(0).getVariableValue(i)] = parents.get(1).getVariableValue(i) ;

```

Comment:
Please factor the `parents.get(x).getVariableValue(i)` to not repeat & search them again every time.